### PR TITLE
Fix/wallet connected status

### LINKED ISSUE #920: 【MantaPay v4】no need to show wallet symbol if wallet refuse to connectd
[Video Link](https://drive.google.com/file/d/1CefbLDNSTI7mfWRjMylQmg0_cEM2G2qH/view?usp=sharing)

### PR DESCRIPTION
**Pull request by jasonintju**
## Description

- close #920 
- fix: wallet connected check method
- fix: give feedback to user when user choose 'reject' or 'cancel'
- update some styles

---

Before we can merge this PR, please make sure that all the following items have been checked off. If any of the checklist items are not applicable, please leave them but write a little note why.

- [ ] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work
- [ ] Re-reviewed files changed in the Github PR explorer

## If PR to `staging`
- [ ] Updated relevant documentation in the code
- [ ] Test code changes manually, and describe your procedure in a comment on this PR
- [ ] Mark this PR with the correct milestone

## If PR to `main`
- [ ] Update the version number in `package.json` and `src/config/common.json`
- [ ] Update the minimum required signer version number in `src/config/common.json` if applicable
- [ ] Test following the procedure in `docs/manual_test.md`
- [ ] Update `CHANGELOG.md`
- [ ] Make sure that all PR's to `manta-signer` and `sdk` on which this PR depends are merged

### CODE REVIEW

File: src/components/Modal/connectWalletModal.tsx [hunk 1]
```diff
@@ -80,10 +81,16 @@ const ConnectWalletBlock = ({
         </a>
       )}
 
-      {walletName === 'Talisman' && !isTalismanExtConfigured && (
-        <p className="absolute left-0 -bottom-5 flex flex-row gap-2 b-0 text-warning text-xsss">
+      {walletConnectingErrorMessages[extensionName] && (
+        <p className="absolute -left-px -bottom-5 flex flex-row gap-2 b-0 text-warning text-xsss">
           <Icon name="information" />
-          You have no account in Talisman. Please create one first.
+          {extensionName === WALLET_NAME.TALISMAN &&
```
Comment (Kevingislason) [inline src/components/Modal/connectWalletModal.tsx:87]: all of this conditional logic on lines 87-92 should probably be handled outside of JSX

File: src/contexts/keyringContext.tsx [hunk 2]
```diff
@@ -46,28 +46,34 @@ type KeyringContextValue = {
 };
 const KeyringContext = createContext<KeyringContextValue | null>(null);
 
+const getInitialWalletConnectingErrorMessages = () => {
```
Comment (jincan39) [inline src/contexts/keyringContext.tsx:49]: Nice job. just wondering whether just Talisman is different and for the other wallets the errorMessage is same?

**Review by jincan39 — APPROVED** [review]

File: src/components/Modal/connectWalletModal.tsx [hunk 3]
```diff
@@ -19,12 +20,24 @@ const ConnectWalletBlock = ({
...
-  const { isTalismanExtConfigured } = useKeyring();
+  const { walletConnectingErrorMessages } = useKeyring();
   const { usingMantaWallet } = useGlobal();
   const { pathname } = useLocation();
   const isCalamariMantaPayPage = pathname.includes('/calamari/transact');
-
   const usingNewUI = usingMantaWallet && isCalamariMantaPayPage;
+  const errorMessage = walletConnectingErrorMessages[extensionName];
+
+  const getDisplayedErrorMessage = () => {
```
Comment (jincan39) [inline src/components/Modal/connectWalletModal.tsx:30]: it always keeps running, should be wrap with useCallback.

Comment (jasonintju) [reply src/components/Modal/connectWalletModal.tsx:30]: About useCallback, only worth it when you are passing down the method as a prop to another component, or the method itself needs to be preserved and not recreated at every render. Optimization always comes with a cost, and most of the time it's not good to use useCallback or useMemo. [cp]

File: src/contexts/keyringContext.tsx (hunk 2 above)
Comment (jasonintju) [reply src/contexts/keyringContext.tsx:49]: Yes, when users have no account in wallet, Talisman's logic is different from others.

**Review by Kevingislason — APPROVED** [review]
Nice work, good idea, this is really useful. The only thing I would suggest is to find a way to get rid of the urls in these error messages, the multi-line errors look a bit awkward. Feel free to do that in a followup pull request.

**Comment by jasonintju** [comment]
> Nice work, good idea, this is really useful. The only thing I would suggest is to find a way to get rid of the urls in these error messages, the multi-line errors look a bit awkward. Feel free to do that in a followup pull request.
> 
> Screen Shot 2023-04-20 at 6 40 54 PM

It will be more clear for users to understand the error message with the urls. `getErrMsgAfterRemovePathname` is to keep the error message in one line as much as possible. The vercel preview url is a bit longer than 'https://app.manta.network'. In prod mode, it will be one line error.

**Pull request merged**